tools/build/ci_install_libpq.sh

```diff
@@ -9,6 +9,13 @@ set -euo pipefail
 set -x
 
 libpq=${1:-}
+rel=$(lsb_release -c -s)
+
+setup_repo() (
+        curl -sL https://www.postgresql.org/media/keys/ACCC4CF8.asc \
+            | sudo tee /etc/apt/trusted.gpg.d/apt.postgresql.org.asc > /dev/null
+        sudo apt-get -qq update
+)
```

Comment:
We can include the source line too here, passing the version as a parameter.

Also, this looks a strange function definition. Shouldn't it use curly brackets (`function NAME { ... }` or `NAME () { ... }`) The body is also over-indented.

As you have simplified the key retrieval, now we don't need `curl | sudo tee` anymore: we can just use `sudo curl`.

```suggestion
setup_repo () {
    version=${1:-}
    sudo curl -sL -o /etc/apt/trusted.gpg.d/apt.postgresql.org.asc \
        https://www.postgresql.org/media/keys/ACCC4CF8.asc
    echo "deb http://apt.postgresql.org/pub/repos/apt ${rel}-pgdg main ${version}" \
        | sudo tee -a /etc/apt/sources.list.d/pgdg.list > /dev/null
    sudo apt-get -qq update
}
```

It should be possible to call it using `setup_repo 10` in the `oldest` branch and leave it as `setup_repo` in `newest`.

Reply:
If we dropped sudo, this would become just:

```suggestion
setup_repo () {
    version=${1:-}
    curl -sL -o /etc/apt/trusted.gpg.d/apt.postgresql.org.asc \
        https://www.postgresql.org/media/keys/ACCC4CF8.asc
    echo "deb http://apt.postgresql.org/pub/repos/apt ${rel}-pgdg main ${version}" \
        >> /etc/apt/sources.list.d/pgdg.list
    apt-get -qq update
}
```

It should be possible to call it using `setup_repo 10` in the `oldest` branch and leave it as `setup_repo` in `newest`.